### PR TITLE
Closes #1913, Closes #1997 Update az_custom_menu_block title style and fix demo content where the card deck page shows menu blocks.

### LINKED ISSUE #1913: Update menu block styles
Follow up to https://github.com/az-digital/az_quickstart/issues/574

1. Apply matching title styles on this menu to match main menu sidebar title styles 
    - Add `class="h4 mt-0"` to rendered `<h2>`
3. ~Make sure that active and visited link styles are still correct in sidebar menus~
    - Not seeing any concerns with active and visited link styles - embedded or sidebar region

### LINKED ISSUE #1997: Card deck and menu block pages inadvertantly share paragraph demo content.
## Problem/Motivation
Go to https://demo.quickstart.arizona.edu/pages/card-decks and see menu blocks embedded in paragraphs instead of card decks.

## Proposed resolution
Determine the correct paragraph ids to use for menu block paragraphs.

### PR DESCRIPTION
**Pull request by trackleft**
## Description
Add classes to the heading for az_custom_menu_block block content blocks by using preprocess field to check if the menu block is being rendered by a az_custom_menu_block and adding some metadata to the menu_block config settings.
Then use hook_preprocess_block to add classes if the metadata exists.

https://813ef390-64a4-407a-8107-decbd65f1f57.probo.build/pages/menu-blocks

## Related issues
Closes #1913
Closes #1997 

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.module [hunk 1]
```diff
@@ -31,3 +31,30 @@ function az_custom_menu_block_form_block_content_az_custom_menu_block_form_alter
     }
   }
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function az_custom_menu_block_preprocess_block(&$variables) {
```
Comment (trackleft) [inline modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.module:38]: Should possibly narrow scope to the sidebar template.

**Review by joeparsons — DISMISSED** [review]
These changes seem OK to me.  Looks like the ProboCI failure was a temporary Probo infrastructure problem or something.

File: modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.module (hunk 1 above)
Comment (joeparsons) [reply modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.module:38]: Seems like a good idea if possible.

File: modules/custom/az_demo/data/az_demo_block_content_block.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
           "settings": {
           "id" : "menu_block:main",
           "label" : "Main navigation",
-          "label_display" : false,
+          "label_display" : true,
```
Comment (bberndt-uaz) [inline modules/custom/az_demo/data/az_demo_block_content_block.json:13]:
```suggestion
          "label_display" : "visible",
```

Although `true` seems to work in order for the title to be displayed, the "Display title" checkbox on the edit page for the custom block is not checked:

image

From testing locally, changing `true` to `"visible"` resulted in the "Display title" checkbox being checked.

File: modules/custom/az_demo/data/az_demo_block_content_block.json [hunk 3]
```diff
@@ -35,7 +35,7 @@
           "settings": {
             "id" : "menu_block:main",
             "label" : "Main navigation",
-            "label_display" : false,
+            "label_display" : true,
```
Comment (bberndt-uaz) [inline modules/custom/az_demo/data/az_demo_block_content_block.json:38]:
```suggestion
            "label_display" : "visible",
```

File: modules/custom/az_demo/data/az_demo_block_content_block.json [hunk 4]
```diff
@@ -61,7 +61,7 @@
...
```
Comment (bberndt-uaz) [inline modules/custom/az_demo/data/az_demo_block_content_block.json:64]:
```suggestion
            "label_display" : "visible",
```

File: modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.module [hunk 5]
```diff
@@ -31,3 +31,30 @@ function az_custom_menu_block_form_block_content_az_custom_menu_block_form_alter
...
+  if ($variables['base_plugin_id'] === 'menu_block') {
+    if (isset($variables['configuration']['az_custom_menu_block']) &&
+        $variables['configuration']['az_custom_menu_block'] === 'az_custom_menu_block'
+    ) {
+      $variables['title_attributes']['class'][] = 'h4';
+      $variables['title_attributes']['class'][] = 'mt-0';
```
Comment (bberndt-uaz) [inline modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.module:44]: Is there any reason not to combine these nested if statements?

```suggestion
  if ($variables['base_plugin_id'] === 'menu_block' && 
    isset($variables['configuration']['az_custom_menu_block']) &&
    $variables['configuration']['az_custom_menu_block'] === 'az_custom_menu_block'
  ) {
    $variables['title_attributes']['class'][] = 'h4';
    $variables['title_attributes']['class'][] = 'mt-0';
```

Comment (trackleft) [reply modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.module]: We could probably remove the base_plugin_id check altogether.

File: modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.module [hunk 6]
```diff
@@ -31,3 +31,30 @@ function az_custom_menu_block_form_block_content_az_custom_menu_block_form_alter
...
   }
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function az_custom_menu_block_preprocess_block(&$variables) {
+  if ($variables['base_plugin_id'] === 'menu_block' && 
+    isset($variables['configuration']['az_custom_menu_block']) &&
+    $variables['configuration']['az_custom_menu_block'] === 'az_custom_menu_block'
```
Comment (trackleft) [inline modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.module:41]:
```suggestion
  if (isset($variables['configuration']['az_custom_menu_block']) &&
      $variables['configuration']['az_custom_menu_block'] === 'az_custom_menu_block'
```

**Review by joeparsons — APPROVED** [review]

**Review by bberndt-uaz — APPROVED** [review]

**Review by danahertzberg — APPROVED** [review]
This is lovely

**Pull request merged**